superset/assets/src/chart/Chart.jsx

```diff
@@ -107,6 +107,14 @@ class Chart extends React.PureComponent {
     const containerStyles = isLoading ? { height, width } : null;
     const isFaded = refreshOverlayVisible && !errorMessage;
     this.renderContainerStartTime = Logger.getTimestamp();
+    if (chartStatus === 'failed') {
+      return (
+        <StackTraceMessage
```

Comment:
I'd put this in a `renderStackTraceMessage()` above then just return that function.

Reply:
noting that you miss the error boundary with this approach